.github/workflows/main.yml

```diff
@@ -0,0 +1,204 @@
+name: CI
+on:
+  push:
+    branches:
+    - auto
+    - try
+  pull_request:
+    branches:
+    - master
+
+jobs:
+  style:
+    name: Check Style
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - name: Install Rust
+      run: rustup update nightly && rustup default nightly
+    - run: ci/style.sh
+
+  docs:
+    name: Build Documentation
```

Comment:
I split this out from the "style" step above because it was taking so long, and I figured the "sequence style before everything" was intended to filter out rustfmt/clippy issues.